### DISCUSSION TITLE
Showing popups over popups

### DISCUSSION THREAD

**Discussion by gktval** [CommunityToolkit/Maui#977] [Q&A]
What is the best way to hide the current popup, show a new popup and wait for the response, then show the old popup again?

Here is what I am currently doing, but the second popup does not show, and it throws a null reference error when it tries to close it.
This code is a bit reduced from what I have, but shows the gist of it.

```
var popup1 = new Popup1();
this.ShowPopup(popup1);
//Do work, get to the point the popup needs to temporarily close
popup1.Close();
//Show second popup and get the user response
answer = await this.ShowPopupAsync(new Popup2());
//Reset the old popup and show it again
popup1.Reset();
popup1.Size = new Size(300, 400);
this.ShowPopup(popup1); //This does not display
//Do more work...
popup1.Close(); This throws an error.
```

**Comment by acaliaro** [comment]
Maybe you could use a single popup and change the content

**Reply by gktval** [reply]
I cannot because information is passed into the second one from elsewhere and for re usability the popup is used elsewhere. Another problem is that the second popup may be 'popup2' or another popup.

**Reply by bijington** [reply]
I need to write this up into an official feature proposal which may help here: https://github.com/CommunityToolkit/Maui/discussions/933

It won't allow you to show a popup that has been shown previously but it would certainly simplify the code to show the same type of popup a second time.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
